test/Microsoft.AspNetCore.Server.KestrelTests/MemoryPoolIteratorTests.cs

```diff
@@ -381,5 +381,66 @@ public void GetsKnownVersion(string input, char endChar, bool expectedResult, st
 
             _pool.Return(block);
         }
+
+        [Theory]
+        [InlineData("HTTP/1.0\r", "HTTP/1.0")]
+        [InlineData("HTTP/1.1\r", "HTTP/1.1")]
+        public void KnownVersionsAreInterned(string input, string expected)
+        {
+            TestKnownStringsInterning(input, expected,
+                (iter, expect) => {
```

Comment:
You can use an extension method as a delegate - that's quite funky; I just learnt something new, thank you @cesarbs :+1: